src/Symfony/Component/Serializer/Normalizer/CollectionDenormalizer.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\PropertyInfo\Type;
+use Symfony\Component\Serializer\Exception\BadMethodCallException;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+
+/**
+ * Denormalizes arrays of objects into a single collection.
+ *
+ * @author Nikita Shipilov <nikita.shipilov.22@gmail.com>
+ *
+ * @final
+ */
+class CollectionDenormalizer implements DenormalizerInterface, DenormalizerAwareInterface, CacheableSupportsMethodInterface
+{
+    use DenormalizerAwareTrait;
+
+    public function denormalize(mixed $data, string $type, string $format = null, array $context = []): \ArrayAccess
+    {
+        if (null === $this->denormalizer) {
+            throw new BadMethodCallException('Please set a denormalizer before calling denormalize()!');
+        }
+
+        $returnType = $this->getCollectionReturnType($type);
+        if (!\is_array($data)) {
+            throw NotNormalizableValueException::createForUnexpectedDataType(sprintf('Data expected to be "%s", "%s" given.', $returnType, get_debug_type($data)), $data, [Type::BUILTIN_TYPE_ARRAY], $context['deserialization_path'] ?? null);
```

Comment:
this should use `array` instead of using the `Symfony\Component\PropertyInfo\Type` constant given that property-info is not a mandatory dependency of the serializer component.